### PR TITLE
Fix: LinearModel Stat

### LINKED ISSUE #3585: add get_out_bias for linear atomic model
Line: 296
https://github.com/deepmodeling/deepmd-kit/blob/e47478ff6adc81cf8051e0856c5067e7bf0a5d7d//deepmd/dpmodel/atomic_model/linear_atomic_model.py#L293-L303

### PR DESCRIPTION
**Pull request by anyangml**
Refactor stat related methods in LinearAtomicModel.
This should resolve #3585 #3586 #3528

### CODE REVIEW

File: deepmd/pt/model/atomic_model/linear_atomic_model.py
Comment (njzjz) [inline deepmd/pt/model/atomic_model/linear_atomic_model.py:1]: This file seems to be the atomic model, not the model.

Comment (njzjz) [reply deepmd/pt/model/atomic_model/linear_atomic_model.py:1]: ping @wanghan-iapcm

**Comment by wanghan-iapcm** [comment]
Should be revised when the #3593 is solved.

File: deepmd/pt/model/atomic_model/linear_atomic_model.py [hunk 1]
```diff
@@ -188,36 +190,46 @@ def forward_atomic(
 
         for i, model in enumerate(self.models):
             mapping = self.mapping_list[i]
+            raw_ret = model.forward_atomic(
```
Comment (wanghan-iapcm) [inline deepmd/pt/model/atomic_model/linear_atomic_model.py:193]: `forward_common_atomic` should be called. You do not need to take care the `apply_out_stat` for each individual model. 
note that the mapped atype is used in `forward_common_atomic`.

**Review by wanghan-iapcm — COMMENTED** [review]
test is required for the application of out_bias on the linearly combined model. 
the other parts of the PR LGTM.

**Review by wanghan-iapcm — APPROVED** [review]

**Review by njzjz — APPROVED** [review]

**Pull request merged**